3p/ampcontext.js

```diff
@@ -24,7 +24,14 @@ export class AmpContext {
    *  @param {Window} win The window that the instance is built inside.
    */
   constructor(win) {
-    this.setupMetadata_();
+    this.win_ = win;
+
+    // if the window sentinel is set directly, that means we are running
+    // in a situation where we couldn't attach the name attribute before
+    // creating the iframe, so don't try to parse the metadata
+    if (!this.getWindowSentinel_()) {
```

Comment:
yep, if that only happens in SafeFrame not NameFrame. If so, update the doc to emphasize that too.